misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -76,60 +80,94 @@ public CoverageReport(Path sourceJarPath, Path targetDirPath, String orgName, St
      * @throws IOException when file operations are failed
      */
     public void generateReport() throws IOException {
-        try {
-            CodeCoverageUtils.unzipCompiledSource(sourceJarPath, projectDir, orgName, moduleName, version);
-        } catch (NoSuchFileException e) {
+
+        // Obtain a path list of all the .jar files generated
+        List<Path> pathList;
+        try (Stream<Path> walk = Files.walk(this.jarCache, 5)) {
```

Comment:
Just for my knowledge, how did we come up with this value as 5 for maxDepth?
Shall we define it as a constant?

Reply:
Defined the Depth as a constant in TesterinaConstants